File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/rust_solutions/2765-longest-alternating-subarray.rs
impl Solution {
    pub fn longest_alternating_subarray(nums: Vec<i32>, threshold: i32) -> i32 {
        let mut max_len = 0;
        let n = nums.len();
        let mut i = 0;

        while i < n {
            if nums[i] > threshold || nums[i] % 2 != 0 {
                i += 1;
                continue;
            }

            let mut current_len = 1;
            let mut j = i + 1;

            while j < n && nums[j] <= threshold && nums[j] % 2 != nums[j - 1] % 2 {
                current_len += 1;
                j += 1;
            }

            max_len = max_len.max(current_len);
            i = j;
        }

        max_len
    }
}

File: leetcode/AIcoding/5.Qwen-3/cpp_solutions/2765-longest-alternating-subarray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int longestAlternatingSubarray(vector<int>& nums) {
        int n = nums.size();
        if (n == 0) return 0;
        int maxLen = 1;
        int currentLen = 1;

        for (int i = 1; i < n; ++i) {
            if ((nums[i] > nums[i - 1] && nums[i - 1] < nums[i - 2]) || 
                (nums[i] < nums[i - 1] && nums[i - 1] > nums[i - 2])) {
                currentLen++;
                maxLen = max(maxLen, currentLen);
            } else {
                currentLen = 1;
            }
        }

        return maxLen;
    }
};